identity/backend/src/main/java/io/camunda/identity/rolemanagement/service/RoleMembershipService.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.identity.rolemanagement.service;
+
+import io.camunda.identity.rolemanagement.repository.RoleRepository;
+import io.camunda.identity.security.CamundaUserDetails;
+import io.camunda.identity.security.CamundaUserDetailsManager;
+import io.camunda.identity.usermanagement.CamundaGroup;
+import io.camunda.identity.usermanagement.service.GroupService;
+import io.camunda.identity.usermanagement.service.UserService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class RoleMembershipService {
+
+  private final UserService userService;
+  private final GroupService groupService;
+  private final CamundaUserDetailsManager camundaUserDetailsManager;
+  private final RoleRepository roleRepository;
+
+  public RoleMembershipService(
+      final UserService userService,
+      final GroupService groupService,
+      final CamundaUserDetailsManager camundaUserDetailsManager,
+      final RoleRepository roleRepository) {
+    this.userService = userService;
+    this.groupService = groupService;
+    this.camundaUserDetailsManager = camundaUserDetailsManager;
+    this.roleRepository = roleRepository;
+  }
+
+  public List<String> getRolesOfUserByUserId(final long userId) {
+    final CamundaUserDetails camundaUserDetails =
+        camundaUserDetailsManager.loadUserByUsername(
+            userService.findUserById(userId).getUsername());
+
+    return camundaUserDetails.getRoles();
+  }
+
+  public void assignRoleToUser(final String roleName, final long userId) {
+    if (!roleRepository.existsById(roleName)) {
+      throw new RuntimeException("role.notFound");
+    }
+
+    final CamundaUserDetails camundaUserDetails =
+        camundaUserDetailsManager.loadUserByUsername(
+            userService.findUserById(userId).getUsername());
+
+    final List<String> roles = new ArrayList<>();
+    roles.addAll(camundaUserDetails.getRoles());
+    roles.add(roleName);
```

Comment:
💭 this can be simplified using streams
```suggestion
    final List<String> roles = Stream.concat(camundaUserDetails.getRoles(), List.of(roleName))
       .collect(Collectors.toList())
```
but, once again, NABD :D

Reply:
Nice idea! Thanks! ... Changed :)